CacheManager.StackExchange.Redis.SelfCleaning/SelfCleaningRedisCacheHandle.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using CacheManager.Core;
+using CacheManager.Core.Internal;
+using CacheManager.Core.Logging;
+using CacheManager.Redis;
+using CacheManager.StackExchange.Redis.SelfCleaning.Core;
+using CacheManager.StackExchange.Redis.SelfCleaning.Timers;
+using StackExchange.Redis;
+
+namespace CacheManager.StackExchange.Redis.SelfCleaning
+{
+    public class SelfCleaningRedisCacheHandle<TCacheValue> : RedisCacheHandle<TCacheValue>, IStartable
+    {
+        private readonly IDatabase _redisDatabase;
+        private readonly ITimer _cleanupTimer;
+        private readonly TimeSpan _slidingExpiration;
+        private readonly int _databaseId;
+        
+        public SelfCleaningRedisCacheHandle(ICacheManagerConfiguration managerConfiguration,
+            CacheHandleConfiguration configuration, ILoggerFactory loggerFactory, ICacheSerializer serializer) : base(
+            managerConfiguration, configuration, loggerFactory, serializer)
+        {
+            RedisConfiguration redisConfiguration = RedisConfigurations.GetConfiguration(configuration.Key);
+
+            if (!(redisConfiguration is SelfCleaningRedisConfiguration selfCleaningRedisConfiguration))
+            {
+                throw new ArgumentException("The given configuration is not of type SelfCleaningRedisConfiguration",
+                    nameof(configuration));
+            }
+
+            _redisDatabase = selfCleaningRedisConfiguration.RedisDatabase;
+            _cleanupTimer = selfCleaningRedisConfiguration.CleanupTimer;
+            _slidingExpiration = selfCleaningRedisConfiguration.SlidingExpiration;
+            _databaseId = selfCleaningRedisConfiguration.Database;
+        }
+
+        public void Start()
+        {
+            _cleanupTimer.Elapsed += RunCleanup;
+            _cleanupTimer.Start();
+        }
+
+        protected override void Dispose(bool disposeManaged)
+        {
+            _cleanupTimer.Elapsed -= RunCleanup;
+            _cleanupTimer.Dispose();
+            
+            base.Dispose(disposeManaged);
+        }
+
+        private void RunCleanup()
+        {
+            IEnumerable<RedisKey> keysToRemove = Servers
+                .SelectMany(server => server.Keys(_databaseId))
+                .Where(key => (_redisDatabase.KeyIdleTime(key) ?? default) >= _slidingExpiration);
```

Comment:
Because the flow isnt trivial (getting the keys from the servers and the IdleTime from the DataBase) You need to add comment here which will explain it

Reply:
Added